aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -0,0 +1,327 @@
+                                int64_t s_q,
+                                int64_t s_kv,
+                                int64_t d,
+                                float scaling_factor,
+                                bool return_softmaxstats,
+                                        bool is_causal,
+                                double dropout_probability,
+                                const Tensor& q,
+                                const Tensor& k,
+                                const Tensor& v,
+                                Tensor& softmaxstats,
+                                Tensor& o,
+                                Tensor& dropoutseed,
+                                Tensor& dropoutoffset,
+                                cudnnHandle_t& handle,
+                                MHAParams& params) {
+    auto dtype = fe::DataType_t::HALF;
+    if (q.scalar_type() == kBFloat16) {
+      dtype = fe::DataType_t::BFLOAT16;
+    }
+    auto mha_graph = std::make_shared<fe::graph::Graph>();
+    mha_graph->set_io_data_type(dtype)
+        .set_intermediate_data_type(fe::DataType_t::FLOAT)
+        .set_compute_data_type(fe::DataType_t::FLOAT);
+
+    auto Q = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                  .set_name("Q")
+                                  .set_dim(std::vector<int64_t>(params.q_dim, params.q_dim+MAX_MHA_DIM))
+                                  .set_stride(std::vector<int64_t>(params.q_stride, params.q_stride+MAX_MHA_DIM)));
+    auto K = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                  .set_name("K")
+                                  .set_dim(std::vector<int64_t>(params.k_dim, params.k_dim+MAX_MHA_DIM))
+                                  .set_stride(std::vector<int64_t>(params.k_stride, params.k_stride+MAX_MHA_DIM)));
+    auto V = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                  .set_name("V")
+                                  .set_dim(std::vector<int64_t>(params.v_dim, params.v_dim+MAX_MHA_DIM))
+                                  .set_stride(std::vector<int64_t>(params.v_stride, params.v_stride+MAX_MHA_DIM)));
+    auto attn_scale = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                       .set_name("attn_scale")
+                                       .set_dim({1, 1, 1, 1})
+                                       .set_stride({1, 1, 1, 1})
+                                       .set_is_pass_by_value(true)
+                                       .set_data_type(fe::DataType_t::FLOAT));
+    //auto bias = mha_graph->tensor(fe::graph::Tensor_attributes()
+    //                         .set_name("bias")
+    //                         .set_dim({b, 1, s_q, s_kv})
+    //                         .set_stride({s_q * s_kv, s_q * s_kv, s_kv, 1}));
+    auto seed = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                     .set_name("Seed")
```

Comment:
Another dumb Q: are these names meaningful? like do they need to match an api for calling the function?